internal/providers/applehv/applehv.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package applehv
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os/exec"
+
+	"github.com/coreos/ignition/v2/config/v3_5_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
+	"github.com/coreos/ignition/v2/internal/platform"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+	"github.com/coreos/vcontext/report"
+	"github.com/mdlayher/vsock"
+)
+
+/*
+	This provider is specific to virtual machines running under Apple Hypervisor on macOS on Apple hardware.
```

Comment:
To be clear, is this mechanism used by any other distro/image on Apple to communicate with the host? E.g. how does it work with cloud-init-based systems?

Reply:
i have no idea how cloud-init and apple's hypervisor work.  I can only do what i am told and bgilbert wanted it this way

Reply:
Ack OK. I can't find any good information either. It seems like the Apple Hypervisor is more of an API than an end-user product. Things like [UTM](https://getutm.app/) which wrap it don't seem to have anything related to cloud-init specifically so I assume it's just done the "old-fashioned way" of attaching an ISO with the user data.